### PR TITLE
k8s: add List API to deployments, cronjobs & statefulsets

### PR DESCRIPTION
**Pull request by rithujohn191**
### Description
This PR adds the following API calls
- Describe & List Deployment
- List CronJob
- List StatefulSet

It also updates the kind script to generate stateful set resources
### Testing Performed
- Unit testing
- Local Testing

**describeDeployment**
```
$ curl -X POST   http://localhost:3000/v1/k8s/describeDeployment -H 'Content-Type: application/json'   -d '{
> "clientset": "kind-clutch-local",
> "cluster": "kind-clutch-local",
> "namespace": "envoy-staging",
> "name": "envoy"
> }'
{"deployment":{"cluster":"kind-clutch-local", "namespace":"envoy-staging", "name":"envoy", "labels":{"app":"envoy"}, "annotations":{"deployment.kubernetes.io/revision":"1"}}
```
**listDeployments**
```
$ curl -X POST 'http://localhost:8080/v1/k8s/listDeployments' \
> -H 'Content-Type: application/json' \
> -d '{
>     "clientset": "kind-clutch-local",
>     "cluster":  "kind-clutch-local",
>     "namespace": "envoy-staging",
>     "options": {}
> }'
{"deployments":[{"cluster":"kind-clutch-local", "namespace":"envoy-staging", "name":"envoy", "labels":{"app":"envoy"}, "annotations":{"deployment.kubernetes.io/revision":"1"}}]}
```
**listCronJobs**
```
$ curl -X POST 'http://localhost:8080/v1/k8s/listCronJobs' -H 'Content-Type: application/json' -d '{
    "clientset": "kind-clutch-local",
    "cluster":  "kind-clutch-local",
    "namespace": "cron-staging",
    "options": {}
}'
{"cronJobs":[{"cluster":"kind-clutch-local", "namespace":"cron-staging", "name":"cron-test", "schedule":"*/1 * * * *", "labels":{"app":"cron"}, "annotations":{"url":"foo@example.com"}}]}
```
**listStatefulSets**
```
$ curl -X POST 'http://localhost:8080/v1/k8s/listStatefulSets' -H 'Content-Type: application/json' -d '{
>     "clientset": "kind-clutch-local",
>     "cluster":  "kind-clutch-local",
>     "namespace": "stateful-staging",
>     "options": {}
> }'
{"statefulSets":[{"cluster":"kind-clutch-local", "namespace":"stateful-staging", "name":"stateful-main", "labels":{}, "annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"apps/v1\",\"kind\":\"StatefulSet\",\"metadata\":{\"annotations\":{},\"name\":\"stateful-main\",\"namespace\":\"stateful-staging\"},\"spec\":{\"replicas\":2,\"selector\":{\"matchLabels\":{\"app\":\"nginx\"}},\"serviceName\":\"stateful-main\",\"template\":{\"metadata\":{\"labels\":{\"app\":\"nginx\"}},\"spec\":{\"containers\":[{\"image\":\"k8s.gcr.io/nginx-slim:0.8\",\"name\":\"nginx\",\"ports\":[{\"containerPort\":80,\"name\":\"web\"}],\"volumeMounts\":[{\"mountPath\":\"/usr/share/nginx/html\",\"name\":\"www\"}]}]}},\"volumeClaimTemplates\":[{\"metadata\":{\"name\":\"www\"},\"spec\":{\"accessModes\":[\"ReadWriteOnce\"],\"resources\":{\"requests\":{\"storage\":\"1Gi\"}}}}]}}\n"}}]}
```

### CODE REVIEW

**Review by danielhochman — APPROVED** [review]
lgtm!

File: tools/kind-stateful-set.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      app: nginx
+  serviceName: "nginx"
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+      - name: nginx
```
Comment (danielhochman) [inline tools/kind-stateful-set.yaml:33]: just curious why nginx?

Comment (rithujohn191) [reply tools/kind-stateful-set.yaml:33]: Just the upstream statefulset example I stole from

Comment (rithujohn191) [reply tools/kind-stateful-set.yaml:33]: It also touches on all the components of a real world statefulset i.e a Headless Service used to control the network domain & the StatefulSet with the nginx container will launch Pods.

Comment (rithujohn191) [reply tools/kind-stateful-set.yaml:33]: If you'd prefer we can just have all our k8s resource run busybox as the container image

Comment (danielhochman) [reply tools/kind-stateful-set.yaml:33]: i have no preference, @mcutalo88 or @scarlettperry can chime in if they feel strongly

Comment (scarlettperry) [reply tools/kind-stateful-set.yaml:33]: fine with me 👍

**Pull request merged**